src/QsCompiler/DataStructures/SyntaxTree.fs

```diff
@@ -342,66 +377,158 @@ type CallableInformation =
 
         { Characteristics = commonCharacteristics; InferredInformation = inferredForAll }
 
+/// The source code range for a type.
+type TypeRange =
+    /// The type is annotated explicitly at the given range.
+    | Annotated of Range
+
+    /// The type is inferred from an expression or other piece of source code at the given range.
+    | Inferred of Range
+
+    /// The type is automatically generated and has no range.
+    | Generated
+
+module TypeRange =
+    /// <summary>
+    /// Creates an <see cref="Annotated"/> type range if a range is given, or the <see cref="Generated"/> type range
+    /// otherwise.
+    /// </summary>
+    let internal annotated = QsNullable<_>.Map Annotated >> QsNullable.defaultValue Generated
+
+    /// <summary>
+    /// Creates an <see cref="Inferred"/> type range if a range is given, or the <see cref="Generated"/> type range
+    /// otherwise.
+    /// </summary>
+    [<CompiledName "Inferred">]
+    let inferred = QsNullable<_>.Map Inferred >> QsNullable.defaultValue Generated
+
+    /// Returns the range if it exists.
+    [<CompiledName "TryRange">]
+    let tryRange =
```

Comment:
nit: TryGetRange to me is a bit more comprehensive than TryRange.

Reply:
Do you mean for the compiled name, the F# name, or both?

Reply:
Both, but just a thought.

Reply:
I don't think that's consistent with F# naming conventions; compare with `List.tryHead`, `List.tryLast`, etc. I generally think "get" is overused in function names since most functions are getting something.